src/compiler/python_generator.cc

```diff
@@ -200,17 +202,16 @@ bool GetModuleAndMessagePath(const Descriptor* type,
     path_elem_type = path_elem_type->containing_type();
   } while (path_elem_type != nullptr);
   string file_name = type->file()->name();
-  string module_name;
   static const int proto_suffix_length = strlen(".proto");
   if (!(file_name.size() > static_cast<size_t>(proto_suffix_length) &&
         file_name.find_last_of(".proto") == file_name.size() - 1)) {
     return false;
   }
-  module_name = file_name.substr(
+  string module = file_name.substr(
```

Comment:
Ideally this could call ModuleName() directly instead of re-implementing it:

https://github.com/google/protobuf/blob/v3.0.0-alpha-1/src/google/protobuf/compiler/python/python_generator.cc#L87

But if that's not possible, it seems better to copy that function verbatim to this file, so it's easier to verify that both contain the same logic.